server/src/test/java/io/crate/integrationtests/UppercaseSchemaNameIntegrationTest.java

```diff
@@ -0,0 +1,94 @@
+import org.junit.Test;
+
+import io.crate.expression.udf.UserDefinedFunctionService;
+import io.crate.expression.udf.UserDefinedFunctionsMetadata;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.view.ViewMetadata;
+import io.crate.metadata.view.ViewsMetadata;
+import io.crate.types.StringType;
+
+public class UppercaseSchemaNameIntegrationTest extends IntegTestCase {
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        var dummyLang = new UserDefinedFunctionsIntegrationTest.DummyLang();
+        Iterable<UserDefinedFunctionService> udfServices = cluster().getInstances(UserDefinedFunctionService.class);
+        for (UserDefinedFunctionService udfService : udfServices) {
+            udfService.registerLanguage(dummyLang);
+        }
+    }
+
+    @Test
+    public void test_using_schema_names_with_uppercases() {
+        execute("create table \"Abc\".t (a int) partitioned by (a)");
+        execute("insert into \"Abc\".t values (1)");
+        execute("create view \"Abc\".v1 as select a from \"Abc\".t");
+        execute("CREATE FUNCTION \"Abc\".func(string) RETURNS STRING LANGUAGE dummy_lang AS 'DUMMY EATS text'");
+        refresh();
+
+        // check index/template names
+        var meta = clusterService().state().metadata();
+        meta.indices().keysIt().forEachRemaining(key -> assertThat(key).startsWith("Abc"));
+        meta.templates().keysIt().forEachRemaining(key -> assertThat(key).startsWith("Abc"));
+
+        // check viewMetadata names as well as its target query
+        ViewsMetadata viewsMetadata = meta.custom(ViewsMetadata.TYPE);
+        ViewMetadata viewMetadata = viewsMetadata.getView(new RelationName("Abc", "v1"));
+        assertThat(viewMetadata).isNotNull();
+        assertThat(viewMetadata.stmt()).isEqualTo(
+            """
+                SELECT "a"
+                FROM "Abc"."t"
+                """
+        );
+
+        // check udfMetadata for proper schema name
+        UserDefinedFunctionsMetadata userDefinedFunctionsMetadata = meta.custom(UserDefinedFunctionsMetadata.TYPE);
+        assertThat(userDefinedFunctionsMetadata.contains("Abc", "func", List.of(StringType.INSTANCE))).isTrue();
+
+        // a little more complex scenario involving schema names with upper cases
+        execute("create table Abc.t (b string, c string as \"Abc\".func(b)) partitioned by (c)");
+        execute("insert into Abc.t(b) values ('Abc')"); // NOTE: here failed before due to - Unknown function: abc.func(abc.t.b)
+        refresh();
+
+        execute("select * from Abc.t");
+        assertThat(printedTable(response.rows())).isEqualTo("Abc| DUMMY EATS text\n");
+
+        assertThatThrownBy(() -> execute("drop function \"Abc\".func(string)"))
+            // TODO: make it more consistent in quoting schema names
+            .hasMessageContaining("Cannot drop function 'Abc.func(text)', it is still in use by 'abc.t.c AS \"Abc\".func(b)'");
```

Comment:
I do not think we are strict on the original form of schema names,
```
cr> select * from A.t;                                                                                           
SchemaUnknownException[Schema 'a' unknown]
cr> select * from "a".t;                                                                                         
SchemaUnknownException[Schema 'a' unknown]
```
Schema names being `Abc` or `"Abc"` should be fine? - `Cannot drop function 'Abc.func(text)', it is still in use by 'abc.t.c AS \"Abc\".func(b)'`

Reply:
Yep that's fine in my opinion.

Reply:
That's fine yes, but overall I still have a conceptul question.\
In my opinion, grammar/lexer should accept `<identifier>` and `"<identifier>"` and then we need to do all the processing ourselves and provide a schema name which is a java string without `"`
-  if the identifier is only letters and no `"` are used, we just lower case the string .e.g: `marios`-> `marios`, `MaRios`->`marios`
- if the identifier has other symbols/digits and no `"` then we throw error, e.g.: `m_ar_ios` or `mar10s`
- If the identifier has `""` then we accept all of its content, provided it's in the accepted chars, and we remove the `""`, e.g.: `"MaRioS"->`MaRioS`, `"_Ma_R_i_O-s"` -> `_Ma_R_i_O-s`

Reply:
> if the identifier is only letters and no " are used, we just lower case the string .e.g: marios-> marios, MaRios->marios

This is already the case and won't change with this PR afaik.

> if the identifier has other symbols/digits and no " then we throw error, e.g.: `m_ar_ios` or `mar10s`

This isn't correct according to the SQL standard. Unquotes identifiers may contain underscores and digits, see https://crate.io/docs/sql-99/en/latest/chapters/02.html#sql-language-identifier. We currently allow this, nothing need to be changed for unquoted identifiers imho.

> If the identifier has "" then we accept all of its content, provided it's in the accepted chars, and we remove the "", e.g.: "MaRioS"->MaRioS, "_Ma_R_i_O-s" -> _Ma_R_i_O-s

This is what this change will do. Our grammar already supported that but we prevent it inside the analysis, not sure what your concerns are related to.

Reply:
Thanks Marios for raising concerns, if anything unresolved yet please let me know.